pkg/cloudprovider/aws/fleet/nodefactory.go

```diff
@@ -36,6 +36,9 @@ type NodeFactory struct {
 func (n *NodeFactory) For(ctx context.Context, instanceIds []*string) ([]*v1.Node, error) {
 	// Backoff retry is necessary here because EC2's APIs are eventually
 	// consistent. In most cases, this call will only be made once.
+	if len(instanceIds) == 0 {
```

Comment:
Curious why you needed to add this? For libraries, I try to avoid "overly defensive" code where every function protects against any possible inputs. Is it possible to just ensure that instanceIds is never empty?

Reply:
If the instanceIds are empty, `DescribeInstances` returns list of all instances in the region. We should not be calling DescribeInstances with an empty list.

Reply:
IMO if `createFleetOutput.Instances` is nil, something has gone wrong. Instead, does it make more sense to include this in your error handling of the CreateFleet response?

if err != nil
if len(createFleetOutput.Errors) != 0
if len(createFleetOutput.Instances) == 0